iree/compiler/Conversion/LinalgToSPIRV/KernelDispatchUtils.cpp

```diff
@@ -63,6 +63,48 @@ static int64_t getMinIfShapeStatic(int64_t shape, int64_t tileSize) {
   return std::min(shape, tileSize);
 }
 
+/// Sets the `tileSizes` and `workgroupSize` for an Linalg `op` to the default,
+/// where at most 3 inner parallel dimensions of `op` are tiled and distributed,
+/// and each invocation handles one scalar elements.
+static LogicalResult setDefaultTilingScheme(
+    const spirv::TargetEnv &targetEnv, linalg::LinalgOp op,
+    TileSizesListType &tileSizes, std::array<int64_t, 3> &workgroupSize) {
+  auto maxWorkgroupSize =
+      targetEnv.getResourceLimits().max_compute_workgroup_invocations();
+
+  const int64_t tileSizeX = 32;
+  const int64_t tileSizeY = maxWorkgroupSize.getInt() / tileSizeX;
+
```

Comment:
I assume you meant to keep things as they were but overall those look like pretty bad default values. Maybe we should have a TODO to revisit that.

Reply:
Yeah. Don't want to change the whole world with one PR. (This PR is already quite large). But +100! Filed https://github.com/google/iree/issues/5852 for this.